mitodl/ocw-studio
# Cc/rootrel with parentage

#### Pre-Flight checklist


#### What are the relevant tickets? https://github.com/mitodl/ocw-studio/issues/1145

#### What's this PR do?
This PR:
- converts `markdown_cleanup rootrel_urls` to use the link parser introduced in #1154 
- changes `markdown_cleanup` management command to skip courses that have *never* been published.
- fix legacy links that link to files with duplicate legacy filenames (the main point of this PR)

#### How should this be manually tested?
1. First, populate `parent_uid`: `docker-compose run --rm web python manage.py overwrite_ocw_course_content --bucket ocw-to-hugo-output-qa --content-field="metadata.parent_uid"` which is needed for this command. See https://github.com/mitodl/ocw-studio/pull/1165/files#diff-c9f4154e325d48a617e2879b73236a5462ed6baa7aa2d2d6503b9106526b8e54R239

    *Note: This populates `metadata.parent_uid` AND sets the `parent_id` FK on WebsiteContent objects, see #1115*
3. Run `docker-compose run --rm web python manage.py markdown_cleanup rootrelative_urls -o rootrel.csv --csv-only-changes` to generate a list of changes that will be made when the command is run in `--comit` mode. Inspect list to makes ure it's reason / links work.

Optionally, run the command in `--commit mode` or without `--csv-only-changes`. If you run the command without `--csv-only-changes`, then the generated CSV will include ALL links on pages that contain ANY rootrelative links. This means that the CSV includes more than just "changes that will be made". In particular, it includes:
1. changes to be made, i.e., rows where `original_text != replacement`. 
2. rows that have not changed, i.e.,`original_text = replacement`. These rows might not have changed for a variety of reasons:
    - the corresponding root-relative links were not broke
    - the corresponding root-relative link was broken but could not be fixed _There are not too many of these_
    - the corresponding link was not a rootrelative link, but was on the same page as a rootrelative link
